src/Uno.UI/Microsoft/UI/Xaml/Controls/TreeView/TreeViewNode.cs

```diff
@@ -0,0 +1,274 @@
+		private object m_itemsSource = null;
+
+		public TreeViewNode()
+		{
+			var collection = new TreeViewNodeVector();
+			collection.SetParent(this);
+			Children = collection;
+			collection.VectorChanged += ChildVectorChanged;
+		}
+
+		public TreeViewNode Parent
+		{
+			get => _parent;
+			private set
+			{
+				_parent = value;
+				UpdateDepth(value != null ? value.Depth + 1 : -1);
+			}
+		}
+
+		public bool HasUnrealizedChildren
+		{
+			get => _hasUnrealizedChildren;
+			set
+			{
+				_hasUnrealizedChildren = value;
+				UpdateHasChildren();
+			}
+		}
+
+		internal bool IsContentMode { get; set; }
+
+		internal TreeNodeSelectionState SelectionState { get; set; }
+
+		public IList<TreeViewNode> Children { get; }
+
+		private void UpdateDepth(int depth)
+		{
+			// Update our depth
+			SetValue(DepthProperty, depth);
+
+			// Update children's depth
+			foreach (var child in Children)
+			{
+				child.UpdateDepth(depth + 1);
+			}
+		}
+
+		private void UpdateHasChildren()
+		{
+			bool hasChildren = Children.Count != 0 || HasUnrealizedChildren;
+			SetValue(HasChildrenProperty, hasChildren);
+		}
+
+		private void ChildVectorChanged(IObservableVector<TreeViewNode> sender, IVectorChangedEventArgs args)
+		{
+			var collectionChange = args.CollectionChange;
+			var index = args.Index;
+			UpdateHasChildren();
+			RaiseChildrenChanged(collectionChange, index);
+		}
+
+		private void OnPropertyChanged(DependencyPropertyChangedEventArgs args)
+		{
+			DependencyProperty property = args.Property;
+			//TODO:
+			//m_propertyChangedEventSource(this, args);
+		}
+
+		private void RaiseChildrenChanged(CollectionChange collectionChange, uint index)
+		{
+			var args = new VectorChangedEventArgs(collectionChange, index);
+			ChildrenChanged?.Invoke(this, args);
+		}
+
+		internal object ItemsSource
+		{
+			get => m_itemsSource;
+			set
+			{
+				//TODO:
+				//m_itemItemsSourceViewChangedRevoker.revoke();
+				m_itemsSource = value;
+				//m_itemsDataSource = value != null ? new ItemsSourceView(value) : null;
+				//if (m_itemsDataSource)
+				//{
+				//	m_itemItemsSourceViewChangedRevoker = m_itemsDataSource.CollectionChanged(auto_revoke, { this, &TreeViewNode.OnItemsSourceChanged });
+				//}
+				//SyncChildrenNodesWithItemsSource();
+			}
+		}
+
+		private void OnItemsSourceChanged(object sender, NotifyCollectionChangedEventArgs args)
+		{
+			switch (args.Action)
+			{
+				case NotifyCollectionChangedAction.Add:
+					{
+						// TreeViewNode and ItemsSource will update each other when data changes.
+						// For ItemsSource -> TreeViewNode changes, m_itemsDataSource.Count() > Children().Size()
+						// We'll add the new node to children collection.
+						// For TreeViewNode -> ItemsSource changes, m_itemsDataSource.Count() == Children().Size()
+						// the node is already in children collection, we don't want to update TreeViewNode again here.
+						if (m_itemsDataSource.Count != Children.Count))
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5156819)